management/WFCORE-1110_Custom_HTTP_Management_Headers.adoc

```diff
@@ -0,0 +1,133 @@
+ * Within the `custom-headers` attribute it will be possible to specify a set of headers using key / value pairs against a specific path.
+ * All paths will be assumed to be prefixes starting with `/` - any paths missing a leading `/` will be handled as though the leading `/` has been specified.
+ * If a specific header is defined multiple times it will be added to the response multiple times.
+ * Configured headers will not override any headers added by the corresponding endpoint.
+
+
+The new attribute to be added to the `/core-service=management/management-interface=http-interface` resource will be called `constant-headers` and will be defined as: -
+
+----
+            "constant-headers" => {
+                "type" => LIST,
+                "description" => "The set of constant HTTP headers to apply to response messages.",
+                "expressions-allowed" => false,
+                "required" => false,
+                "value-type" => {
+                    "path" => {
+                        "type" => STRING,
+                        "description" => "The path the headers should be applied to.",
+                        "expressions-allowed" => true,
+                        "required" => true,
+                    },
+                    "headers" => {
+                        "type" => LIST,
+                        "description" => "The headers to set for the matched path.",
+                        "expressions-allowed" => false,
+                        "required" => false,
+                        "value-type" => {
+                            "header" => {
+                                "type" => STRING,
+                                "description" => "The name of the HTTP header to set.",
+                                "expressions-allowed" => true,
+                                "required" => true,
+                            },
+                            "value" => {
+                                "type" => STRING,
+                                "description" => "The value to set for the HTTP header.",
+                                "expressions-allowed" => true,
+                                "required" => true,
+                            }
+                        }
+                    }
+                }
+----
+
+The following example hows how the interface can be configured to add a custom header for the `/management` endpoint: -
+
+----
+/] /core-service=management/management-interface=http-interface:write-attribute(name=constant-headers, value=[{path=/management, headers=[{header=X-Header, value=HeaderValue}]}])
+----
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+As the contexts supported by the HTTP management interface can be added dynamically by subsystems no validation will be performed to verify that the specified paths are genuinely reachable paths as this information is not available to us at the time we perform model validation.
+
+== Test Plan
+
```

Comment:
I am currently preparing the test plan, however some high-level description of the test plan should be added here as well.

Reply:
I will add a little bit for the tests I am adding to WildFly Core but feel free to send a PR against this branch if you would like to add some content to this block https://github.com/darranl/wildfly-proposals/tree/WFCORE-1110